ember-server/src/test/scala/org/http4s/ember/server/EmberServerMtlsSuite.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.server
+
+import cats.effect._
+import fs2.io.net.tls.TLSContext
+import fs2.io.net.tls.TLSParameters
+import java.net.URL
+import java.nio.charset.StandardCharsets
+import java.security.KeyStore
+import javax.net.ssl._
+import org.http4s._
+import org.http4s.server.Server
+import org.http4s.implicits._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.server.ServerRequestKeys
+import org.http4s.testing.ErrorReporting
+import scala.io.Source
+import scala.util.Try
+
+/** Test cases for mTLS support in Ember Server.
+  */
+class EmberServerMtlsSuite extends Http4sSuite {
+  {
+    val hostnameVerifier: HostnameVerifier = new HostnameVerifier {
+      override def verify(s: String, sslSession: SSLSession): Boolean = true
+    }
+
+    // For test cases, don't do any host name verification. Certificates are self-signed and not available to all hosts
+    HttpsURLConnection.setDefaultHostnameVerifier(hostnameVerifier)
+  }
```

Comment:
Is all of this really necessary? I'm no expert on this by any means but fs2 tests TLS without any of this. https://github.com/typelevel/fs2/blob/main/io/jvm/src/test/scala/fs2/io/net/tls/TLSSocketSuite.scala#L126

Reply:
I removed this code and the tests passed, so it looks like this isn't necessary. I'll push an update for this soon.

Reply:
I updated the comment above - not sure why things seemed to work before, but they clearly didn't when I tested just now. I got the error that was mentioned above.

Reply:
BTW we could add a `fromKeyStoreAndTrustStore` method to `TLSContextPlatform.scala`

Reply:
If I rewrite the tests as described above, I can get rid of this code. My previous comment was made with the assumption that the old implementation was going to stay.

Reply:
> BTW we could add a `fromKeyStoreAndTrustStore` method to `TLSContextPlatform.scala`
It would be useful in this case - One thing I notice about `fromKeyStoreResource` is that the various getInstance methods only allow the usage of the default types and algorithms (that is, the defaults are hard-coded):
```
KeyStore.getInstance(KeyStore.getDefaultType)
KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
```
So I don't think I could use this with a different keystore type such as a LunaHSM provider, or whitecryption SKB provider. Not sure if that's going beyond the scope of what's intended here.